### PR TITLE
Adds clothing production for Botany

### PR DESCRIPTION
**Pull request by MMMiracles**
:cl: MMMiracles
add: Botany can now grow cotton to produce cloth for various jumpsuits.
add: Cotton can be mutated into a much more durable strand, allowing for the production of crude armor.
add: Rainbow flowers, an alternative for clothing dyeing, is obtainable through cargo's exotic seed crate.
add: A loom can now be crafted with some planks so you can actually weave that cotton into a usable fabric.
add: Fannypacks, softcaps, beanies and scarves are now craftable with cloth and dyeable.
/:cl:

[why]: A way to produce clothing and naturally obtain cloth past ripping up other stuff seemed useful and with the economy aspect, might actually have some extra use.

**Things To Do**
- [x] Durathread sprites
- [x] Rainbow Flower sprites
- [x] Make washing machine actually accept the flowers as a proper dye source
- [x] Crafting recipes for durathread
- [x] Balance the plant stats

flowers

The loom sprite is codersprite-tier please someone find a replacement for this it hurts to look at.

### CODE REVIEW

**Comment by MMMiracles** [comment]
spaceman doesn't have the finesse to weave thousands of single strands into a usable fabric by themselves.

**Comment by Qustinnus** [comment]
https://www.youtube.com/watch?v=-iTJPzp27-s this pr

**Comment by PKPenguin321** [comment]
I humbly request you place cotton seeds in the public garden on the maps as well

**Comment by FantasticFwoosh** [comment]
Atomise this please OP; the Autolathe is very well known to be a mess of :spaghetti: that would need a refactor. Trust me i've tried and i believe autolathes won't go anywhere in the future without a code simplification and overhaul.

- If you can pull that off, i have autodrobe sprites from my own project attempt that to give you to speed things along.

Otherwise I think just focusing on the cotton/loom interactions would be best and scratching out your other checklists temporarily. Why would the dye plants even exist without first stabilising the methods you get the cloth.

- Besides plenty of questions like: do you dye the cloth or the clothes first? Couldn't you make washing machines actually constructible similar to a dish drive for this purpose etc.

**Comment by Barhandar** [comment]
we rimworld now

Don't variant autolathe, make it a self-use crafting menu same as you can use _already existing cloth_ for. The only machine in the process should be either loom (craftable out of wood) or none whatsoever; otherwise there's already biogenerator that produces cloth out of junk.
>spaceman doesn't have the finesse to weave thousands of single strands into a usable fabric by themselves.

Doesn't matter, it's a game where you can grow magic healing wormwood that turns your hydroponics trays into infinite matter producers, eat "nutriment" which fulfills 100% of your nutritional needs, and you have a machine that turns the same amount of energy into equal volumes of _vastly_ different-weight chemicals.

**Comment by BebeYoshi** [comment]
@Barhandar I may be wrong but @FantasticFwoosh mentioned something about they are planning to remove biogenerator, but I still agree with the crafting menu thing
About the plants stats, I suggest you change the maturation speed of the rainbow flower to 6 and raises the production speed to 2-5 (between both of these) also lower their potency 20 like other flower and yield 4 or 5, endurance to 10 and lifespan around 20 and it should be fine, the cotton stats seems pretty ok for me

**Comment by FantasticFwoosh** [comment]
Im not planning to remove the biogenerator, its just at the moment relevant to my own PR that's up **the maintainers** are looking into removing it and stalling PR's.

**Comment by ExcessiveUseOfCobblestone** [comment]
> refactor the biogen's list of contents

pardon?

Also just have the loom be the crafting station for all things clothes, that way you're still "gated" by needing the item without also having to go 500 places for a jumpsuit.

**Comment by epochayur** [comment]
fucking ace

👍

**Comment by Barhandar** [comment]
>Also just have the loom (with table-crafting) be the crafting station for all things clothes, that way you're still "gated" by needing the item without also having to go 500 places for a jumpsuit.

can we get sheetcrafting that requires loom in proximity instead of the tablecrafting menu

**Comment by MMMiracles** [comment]
I'm dropping the autolathe idea as a part 2 to this, since that'll require adding material value to a lot of clothing items and stuff. 

All those complaining about the loom seem to of forgot that leather has an entire tanning process to go through before you can actually use it.

**Comment by dr-bee** [comment]
Any chance you could make the rainbow flowers contain the corresponding color of crayon powder?

**Comment by Barhandar** [comment]
>All those complaining about the loom seem to of forgot that leather has an entire tanning process to go through before you can actually use it.

And that is terrible and useless and should be gone. Bonus: it only applies to handmade leather from mobs.

**Comment by DaxDupont** [comment]
Giving armor to botany what could go wrong

**Comment by Kagus** [comment]
> Giving armor to botany what could go wrong

Giving citizenship to cats what could go wrong

**Comment by ExcessiveUseOfCobblestone** [comment]
@Barhandar there's a clothing subsect in TC already so I don't really see the issue of it being TC, particularly when the tool aspect is already incorporated into the mechanic.

**Comment by Barhandar** [comment]
I see the issue of it being TC due to TC being fairly awkward to use.

**Comment by MMMiracles** [comment]
is good for review now

**Comment by MMMiracles** [comment]
i lied now it is

File: code/modules/clothing/head/helmet.dm [hunk 1]
```diff
@@ -243,6 +243,14 @@
 	item_state = "skull"
 	strip_delay = 100
 
+/obj/item/clothing/head/helmet/durathread
+	name = "makeshift helmet"
+	desc = "A hardhat with strips of leather and durathread for additional blunt protection."
+	icon_state = "durathread"
+	item_state = "durathread"
+	armor = list("melee" = 25, "bullet" = 10, "laser" = 20,"energy" = 10, "bomb" = 30, "bio" = 15, "rad" = 20, "fire" = 100, "acid" = 50)
```
Comment (FantasticFwoosh) [inline code/modules/clothing/head/helmet.dm]: For comparison, this is **_more powerful_** than a skull helmet made out of just bones versus fire, indirect bomb damage & acid while the skull helmet compensates for more rounded statistics versus more conventional damage types.

- More hardhats can be found in the 'engineering supplies' cargo crate to dismiss limited supply reasoning & concerns for crafting, at up to 3 hats a crate.

Comment (MMMiracles) [reply code/modules/clothing/head/helmet.dm]: those stats are also from the hard hat themselves so

**Comment by optimumtact** [comment]
Someone needs to add recipes to make all the exotic clothing in the vendors so botany can undercut the station

**Pull request merged**